apps/kitchensink-react/src/DocumentCollection/MultiResourceRoute.tsx

```diff
@@ -1,110 +1,152 @@
+      />
+      {author?.role && <p style={{color: '#fff', marginBottom: '1rem'}}>Role: {author.role}</p>}
+      {author?.awards && author.awards.length > 0 && (
+        <div>
+          <h4 style={{marginBottom: '0.5rem'}}>Awards</h4>
+          <ul style={{paddingLeft: '1.5rem'}}>
+            {author.awards.map((award: string, index: number) => (
+              <li key={index}>{award}</li>
+            ))}
+          </ul>
+        </div>
+      )}
+    </div>
+  )
+}
+
+function PlayerEditor({docHandle}: {docHandle: DocumentHandle<'player'>}) {
+  const {data: player} = useDocument(docHandle)
+  const setPlayerName = useEditDocument({...docHandle, path: 'name'})
+
+  return (
+    <div
+      style={{
+        padding: '2rem',
+        borderRadius: '8px',
+        backgroundColor: '#b4e6ef',
+        boxShadow: '0 2px 4px rgba(0,0,0,0.1)',
+        minWidth: '300px',
+        flex: 1,
+        color: '#444',
+      }}
+    >
+      <h2 style={{marginBottom: '1rem', color: '#2a2a2a'}}>
+        Player Document ({docHandle.projectId}.{docHandle.dataset})
+      </h2>
+      <a
+        style={{display: 'block', marginBottom: '1rem', color: '#3e41e7'}}
+        href={`https://autofoos.com/structure/player;${player?._id}`}
+        target="_blank"
+        rel="noopener noreferrer"
+      >
+        View in Studio →
+      </a>
+
+      <h3 style={{fontSize: '1.5rem', marginBottom: '0.5rem'}}>{player?.name}</h3>
+      <TextInput
+        label="Name"
+        type="text"
+        value={player?.name}
+        onChange={(e) => setPlayerName(e.currentTarget.value)}
+      />
+      <div style={{color: '#444', marginBottom: '1rem'}}>
+        {player?.slackUserId && `Slack User ID: ${player.slackUserId}`}
+      </div>
+    </div>
+  )
+}
 
 export function MultiResourceRoute(): JSX.Element {
-  const {data: author} = useDocument(doc)
-  const {data: player} = useDocument(doc2)
-  const setAuthorName = useEditDocument({...doc, path: 'name'})
-  const setPlayerName = useEditDocument({...doc2, path: 'name'})
+  const configs = import.meta.env['VITE_IS_E2E'] ? e2eConfigs : devConfigs
+  const {data: authorDocuments} = useDocuments({
+    documentType: 'author',
+    batchSize: 1,
+    projectId: configs[0].projectId,
+    dataset: configs[0].dataset,
+  })
+
+  const {data: playerDocuments} = useDocuments({
+    documentType: 'player',
+    batchSize: 1,
+    projectId: configs[1].projectId,
+    dataset: configs[1].dataset,
+  })
+
+  const [authorHandle, setAuthorHandle] = useState<DocumentHandle<'author'> | null>(null)
+  const [playerHandle, setPlayerHandle] = useState<DocumentHandle<'player'> | null>(null)
+
+  useEffect(() => {
+    if (authorDocuments.length > 0) {
+      const doc = authorDocuments[0]
+      setAuthorHandle(doc)
+    }
+  }, [authorDocuments])
+
+  useEffect(() => {
+    if (playerDocuments.length > 0) {
+      const doc = playerDocuments[0]
+      setPlayerHandle(doc)
+    }
+  }, [playerDocuments])
```

Comment:
Also curious about `useState`/`useEffect` here but guessing the answer is the same as the previous will be! ;)

Reply:
none reasons except old habit. fixed now, thanks